Shared styles

ianmclean2011 opened the pull request:
Two things I want to point out:
1) The issue with font-family: inherit (just want to make sure that the comment is sufficient, and that how I'm handling it is okay)
2) I think the d2l-typography-shared-styles.html link is unnecessary for both the main d2l-text-input.html file, and the shared styles file ... just want to confirm that is correct (I can make the change to remove if that is the case)

dlockhart commented on d2l-text-input.html:
```diff
@@ -18,61 +20,36 @@
-				vertical-align: middle;
-				width: 100%;
+				@apply --d2l-input;
+				@apply --d2l-input-disabled-hover;
```
Question (because I'm not sure): can mixins be applied from within mixins? If so, we could apply `--d2l-input-disabled-hover` from inside `--d2l-input` so that other people consuming these wouldn't need to know about that subtlety.

ianmclean2011 commented:
It looks like I can use a mixin in a mixin. I moved the --d2l-input-disabled-hover into --d2l-input, and it seems to work fine (I inspected the element, and can see the extra styles being added). Also removed the typography shared styles, and moved typography to the dev-dependencies. Everything seems to look and behave the same in the demo, so it looks good.
